### ISSUE TITLE
Splash Attention is Broken on TPU Pods and does not follow keras.config.disable_flash_attention()

### ISSUE THREAD

**Issue by chaosmaster142857** [keras-team/keras#21116]
**Describe the bug**
- Splash Attention breaks many small keras hub models on tpuv4-pods when using DataParallel. For instance I have tested a few siglip models and the clip model, and the same thing still happens.

**To Reproduce**
[Colab Link](https://colab.research.google.com/drive/1CyId_6MoGMEWQrk8kQJpo7zDnK64qOkW?usp=sharing)

**Expected behavior**
The model to not crash with an error and to properly run the siglip model:
Keras:
20.6% that image 0 is 'This is a photo of 2 cats'
0.0% that image 1 is 'This is a photo of 2 dogs'
Loss: 6.132134437561035

**Additional context**
- This is reproducible with tpuv4-64 (and similar sizes) and tpuv2-8 (colab)
- I have tried the fix from: https://github.com/jax-ml/jax/pull/27461, but it seems to give me a different error instead. It isn't merged yet, so changes may still happen.

**Would you like to help us fix it?**
- Frankly, I'm not sure how to fix the splash attention implementation in jax's pallas, so I think that having a way to disable it for tpus and use normal attention would be helpful.
- I was able to get it running by using normal attention in jax:

 return jax.nn.dot_product_attention(
            query,
            key,
            value,
            bias=bias,
            mask=mask,
            scale=scale,
            is_causal=is_causal,
            implementation= "xla",
        ) 

before https://github.com/keras-team/keras/blob/6d26efbe97c72323566ef12ca639a888ab3ea340/keras/src/backend/jax/nn.py#L1181
This just returns using normal flash attention.

**Comment by sonali-kumari1** [comment]
Hi @chaosmaster142857 - 

I have tested your code with v2-8 TPU(colab) and was unable to reproduce the crash. The siglip model executed properly with the following output:
```
Keras:
20.5% that image 0 is 'This is a photo of 2 cats'
0.0% that image 1 is 'This is a photo of 2 dogs'
Loss: 6.136449337005615
```
I am attaching a [gist](https://colab.sandbox.google.com/gist/sonali-kumari1/1ad8b7270d8069867a4469c033a37dc3/21116.ipynb) for your reference. Thanks!

**Comment by sonali-kumari1** [comment]
Hi @chaosmaster142857, Please feel free to close this issue. Thanks!
